### PR TITLE
plug in LedgerWindow

### PR DESCRIPTION
**Pull request by rob-solana**

### CODE REVIEW

**Comment by rob-solana** [comment]
the depths to which I had to plumb options and &mut Option<&mut LedgerWindow> makes me think there's a better way

**Comment by rob-solana** [comment]
the large network test takes longer to converge and does so less often.  I think this is because a disk seek and read in the gossip pipeline is hurting us.  I think repair requests from disk may need their own stage/thread, whatever.

File: src/crdt.rs [hunk 1]
```diff
@@ -954,6 +955,7 @@ impl Crdt {
     }
     fn run_window_request(
         window: &SharedWindow,
+        ledger_window: &mut Option<&mut LedgerWindow>,
```
Comment (garious) [inline src/crdt.rs]: Why is the `Option` `&mut`?

Comment (rob-solana) [reply src/crdt.rs]: it was the only way I could figure out how to get the ledgerwindow through a move and a loop

Comment (rob-solana) [reply src/crdt.rs]: if you're on the Rust server, you can look up my question in #beginners

**Pull request merged**